### PR TITLE
Enhance mobile view

### PR DESCRIPTION
**Pull request by alex-ju**
This PR updates the way miller-columns-element is behaving on mobile/small screens.

**Chore**
- Update `govuk-frontend` to the latest version (3.3.0)
- Watch missing SCSS file

**Changes**
- Mark the active column with a CSS modifier (and use that to only show the active column on mobile)
- Add a heading to each column (on mobile) to show context
- Add back button (on mobile) to allow upstream navigation
- Update dist to reflect changes

<table>
<tr><th>Before</th><th>After</th></tr>
<tr><td>

Screenshot 2019-11-26 at 10 38 03

</td><td>

Screenshot 2019-11-26 at 10 37 19

</td></tr>
</table>

[Trello card](https://trello.com/c/ApHlPUDQ)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -293,6 +293,7 @@ class MillerColumnsElement extends HTMLElement {
       columnCollapse: 'miller-columns__column--collapse',
       columnMedium: 'miller-columns__column--medium',
       columnNarrow: 'miller-columns__column--narrow',
+      columnActive: 'miller-columns__column--active',
```
Comment (ChrisBAshton) [inline index.js]: This code looks good, but could the commit be split into two?

>Identify the active column based on the active item and refactor column styling to only show the active column, full width, on mobile devices.

=>

1) `Identify the active column based on the active item`
2) `Refactor column styling to only show the active column, full width, on mobile devices`

File: index.js [hunk 2]
```diff
@@ -487,6 +497,11 @@ class MillerColumnsElement extends HTMLElement {
         // show this column in all it's glory
         item.classList.remove(collapseClass, narrowClass, mediumClass)
       }
+
+      // mark column with active item as active
```
Comment (ChrisBAshton) [inline index.js]:
```suggestion
      // mark last column as active
```

File: index.js [hunk 3]
```diff
@@ -467,6 +474,9 @@ class MillerColumnsElement extends HTMLElement {
         } else if (showNarrow) {
           item.classList.add(narrowClass)
         }
+        if (columnsToShow.length === 0) {
```
Comment (ChrisBAshton) [inline index.js]: This looks like it's covering an edge case - probably best moved to its own commit.

File: index.js [hunk 4]
```diff
@@ -327,6 +328,23 @@ class MillerColumnsElement extends HTMLElement {
     const div = document.createElement('div')
 
     if (!root) {
+      // Append back link
+      const backLink = document.createElement('button')
+      backLink.className = this.classNames.backLink
+      backLink.type = 'button'
+      backLink.innerHTML = 'Back'
```
Comment (ChrisBAshton) [inline index.js]: As far as I know, this element is i18n friendly until we add this `Back` button - now we have hardcoded English in the JS. Can it be read from a parameter in the `<miller-columns-selected` invocation, perhaps defaulting to `Back` if not provided?

File: test/test.js [hunk 5]
```diff
@@ -261,6 +261,19 @@ describe('miller-columns', function() {
...
+      const firstColumn = document.querySelectorAll('.miller-columns__column')[0]
+      const firstItemL1 = firstColumn.querySelector('li')
+      const secondColumn = document.querySelectorAll('.miller-columns__column')[1]
+      const backButtonL2 = secondColumn.querySelector('.govuk-back-link')
+
+      firstItemL1.click()
+      assert.equal(document.querySelector('.miller-columns__column--active'), secondColumn)
+
+      backButtonL2.click()
+      assert.equal(document.querySelector('.miller-columns__column--active'), firstColumn)
```
Comment (ChrisBAshton) [inline test/test.js]: Nice test 👍

File: examples/checkboxes-checked.html [hunk 6]
```diff
@@ -2,6 +2,7 @@
 <html lang="en" class="govuk-template">
 <head>
   <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">
```
Comment (ChrisBAshton) [inline examples/checkboxes-checked.html]: The meta tag on GOV.UK is simply `<meta name="viewport" content="width=device-width, initial-scale=1">` - is there a reason why we want `viewport-fit=cover` too?

File: index.js (hunk 4 above)
Comment (alex-ju) [reply index.js]: i18n was not on the list of things considered at the initial build. This is not the only case where we hardcode content we do it for the "Remove topic" button and "No selected topics" message. So I wouldn't extract it as part of this work, but I agree it's a potential improvement that can be done across the component.

File: index.js (hunk 3 above)
Comment (alex-ju) [reply index.js]: Extracted this into its own commit.

File: index.js (hunk 2 above)
Comment (alex-ju) [reply index.js]: I was skeptical about naming it 'last column' as, depending on your interaction, it may not be the last one in the chain – but whatever makes more sense to everyone.

File: index.js (hunk 1 above)
Comment (alex-ju) [reply index.js]: Split this into separate commits as suggested.

File: examples/checkboxes-checked.html (hunk 6 above)
Comment (alex-ju) [reply examples/checkboxes-checked.html]: I added this based on the GOV.UK template file but for this specific example, it's not needed indeed so I removed it.

**Comment by alex-ju** [comment]
Thank you for the review @ChrisBAshton – I tried to address all your comments. It should be good for a new round.

File: index.js (hunk 2 above)
Comment (ChrisBAshton) [reply index.js]: "Last visible column" maybe?

File: index.js (hunk 4 above)
Comment (ChrisBAshton) [reply index.js]: 👍 cool, didn't realise there was already hardcoded English in the codebase.

File: examples/checkboxes-checked.html (hunk 6 above)
Comment (ChrisBAshton) [reply examples/checkboxes-checked.html]: Doesn't matter either way IMO - I'm more interested in why the same meta value hasn't been applied to the homepage :)

File: index.js (hunk 3 above)
Comment (ChrisBAshton) [reply index.js]: Thanks - that makes perfect sense to me now.

File: index.js [hunk 7]
```diff
@@ -474,6 +474,9 @@ class MillerColumnsElement extends HTMLElement {
         } else if (showNarrow) {
           item.classList.add(narrowClass)
         }
+        if (columnsToShow.length === 0) {
+          item.classList.add(activeClass)
```
Comment (ChrisBAshton) [inline index.js]: Probably worth writing a unit test for this edge case too 👍

File: index.js (hunk 1 above)
Comment (ChrisBAshton) [reply index.js]: Thanks.

File: examples/checkboxes-checked.html (hunk 6 above)
Comment (alex-ju) [reply examples/checkboxes-checked.html]: The GOV.UK homepage uses a deprecated layout at the moment. Shall be updated – whenever resources are dedicated to this.

File: index.js (hunk 7 above)
Comment (alex-ju) [reply index.js]: Mmm, it's captured already, [checking the first column is active by default](https://github.com/alphagov/miller-columns-element/pull/25/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R250).

Comment (ChrisBAshton) [reply index.js]: Ideally that line of code n the test would be added in the same commit as this one 👍 as otherwise I imagine the tests would have been failing at that point.

Comment (alex-ju) [reply index.js]: Agree, when I split the commit I should've split the test too. I now moved the assertion in the appropriate commit.

File: index.js (hunk 2 above)
Comment (alex-ju) [reply index.js]: Updated with the suggestion.
